Dockerfiles/gadget.Dockerfile

```diff
@@ -20,7 +22,8 @@ RUN cd /gadget && go mod download
 # This COPY is limited by .dockerignore
 COPY ./ /gadget
 RUN cd /gadget/gadget-container && \
-	make -j$(nproc) TARGET_ARCH=${TARGETARCH} gadget-container-deps
+	make -j$(nproc) TARGET_ARCH=${TARGETARCH} gadget-container-deps \
+	-ldflags"-X github.com/inspektor-gadget/inspektor-gadget/internal/version.version=${VERSION}"
```

Comment:
-ldflags has to be passed to the `go build` command, not to the makefile. Something like 

```diff
$ git diff gadget-container/
diff --git gadget-container/Makefile gadget-container/Makefile
index b9a1a65e1..14763b8bd 100644
--- gadget-container/Makefile
+++ gadget-container/Makefile
@@ -3,6 +3,7 @@ gadget-container-deps: entrypoint cleanup ocihookgadget gadgettracermanager nrig
 
 
 TARGET_ARCH ?= $(shell go env GOHOSTARCH)
+VERSION ?= $(shell git describe --tags --always --dirty)
 
 # Entrypoint
 
@@ -26,6 +27,7 @@ cleanup:
 gadgettracermanager:
        mkdir -p bin
        GO111MODULE=on CGO_ENABLED=0 GOOS=linux GOARCH=$(TARGET_ARCH) go build \
+               -ldflags "-X github.com/inspektor-gadget/inspektor-gadget/internal/version.version=$(VERSION)" \
                -o bin/gadgettracermanager \
                ./gadgettracermanager/
``` 

could work. (after removing this line)

Reply:
@mauriciovasquezbernal `VERSION ?= $(shell git describe --tags --always --dirty)` does this set version to the commit hash? Why would we do that?

Reply:
`git describe` provides a human readable description of the state of the repo. See `git help describe`. We already use that in the main Makefile: 

https://github.com/inspektor-gadget/inspektor-gadget/blob/e38a90935da28cd56abca71a1fc01eb4e779e272/Makefile#L1